src/table/cells/__tests__/handle-key-press.spec.js

```diff
@@ -0,0 +1,85 @@
+import { arrowKeysNavigation, getRowAndColumnCount } from '../handleKeyPress';
+
+const { JSDOM } = require('jsdom');
+
+describe('handle-key-press', () => {
+  describe('moveToNextFocus', () => {});
+
+  describe('arrowKeysNavigation', () => {
+    const evt = {};
+    const rowAndColumnCount = {};
+    let rowIndex;
+    let colIndex;
+
+    it('should stay the current cell', () => {
+      evt.key = 'ArrowDown';
+      rowAndColumnCount.rowCount = 1;
+      rowAndColumnCount.columnCount = 1;
+      rowIndex = 0;
+      colIndex = 0;
+      const { nextRow, nextCol } = arrowKeysNavigation(evt, rowAndColumnCount, rowIndex, colIndex);
+      expect(nextRow).to.eql(0);
+      expect(nextCol).to.eql(0);
+    });
+
+    it('should go to one row down cell', () => {
+      evt.key = 'ArrowDown';
+      rowAndColumnCount.rowCount = 2;
+      rowAndColumnCount.columnCount = 1;
+      rowIndex = 0;
+      colIndex = 0;
+      const { nextRow, nextCol } = arrowKeysNavigation(evt, rowAndColumnCount, rowIndex, colIndex);
+      expect(nextRow).to.eql(1);
+      expect(nextCol).to.eql(0);
+    });
+
+    it('should go to one row up cell', () => {
+      evt.key = 'ArrowUp';
+      rowAndColumnCount.rowCount = 2;
+      rowAndColumnCount.columnCount = 1;
+      rowIndex = 1;
+      colIndex = 0;
+      const { nextRow, nextCol } = arrowKeysNavigation(evt, rowAndColumnCount, rowIndex, colIndex);
+      expect(nextRow).to.eql(0);
+      expect(nextCol).to.eql(0);
+    });
+
+    it('should go to one column left cell', () => {
+      evt.key = 'ArrowLeft';
+      rowAndColumnCount.rowCount = 1;
+      rowAndColumnCount.columnCount = 2;
+      rowIndex = 0;
+      colIndex = 1;
+      const { nextRow, nextCol } = arrowKeysNavigation(evt, rowAndColumnCount, rowIndex, colIndex);
+      expect(nextRow).to.eql(0);
+      expect(nextCol).to.eql(0);
+    });
+
+    it('should go to one column right cell', () => {
+      evt.key = 'ArrowRight';
+      rowAndColumnCount.rowCount = 1;
+      rowAndColumnCount.columnCount = 2;
+      rowIndex = 0;
+      colIndex = 0;
+      const { nextRow, nextCol } = arrowKeysNavigation(evt, rowAndColumnCount, rowIndex, colIndex);
+      expect(nextRow).to.eql(0);
+      expect(nextCol).to.eql(1);
+    });
+  });
+
+  describe('getRowAndColumnCount', () => {
+    let resolvedRowCount;
+    let resolvedColumnCount;
+
+    it('should return a rowElements, rowCount, and columnCount', () => {
+      resolvedRowCount = 15;
+      resolvedColumnCount = 2;
+      JSDOM.fromFile('src/table/cells/__tests__/table.html').then((dom) => {
+        const rootElement = dom.window.document;
```

Comment:
I'm not sure about this. One of the perks of isolating the `getRowAndColumnCount` function is that you can pass a mocked `rootElement` with mocked functions on it, returning some number of rows/columns then checking if `getRowAndColumnCount` returns the same

Reply:
I think 'passing a mocked rootElement with a mocked method on it, returning some number of rows/columns then checking if getRowAndColumnCount returns the same' is what I did here.

Reply:
Yes, but you don't need to create a whole DOM. We know what `getElementsByClassName` should return, given a certain DOM. You can just create an object with the `getElementsByClassName` mocked returning an array of "elements", the same way the real method would

Reply:
I think that if I create an object - rootElement, with a mocked getElementsByClassName which returns rowElements. Then it only really test the length in the getRowAndColumnCount function.

Reply:
Well, not only that. The returned object should also contain the same thing as `getElementsByClassName` returns, plus both of the lengths of the arrays. Adding a whole DOM for a unit test, then it becomes a bit more of an integration test or something. Since this feature does a lot of DOM manipulations we should have integration tests for keyboard navigation to complement the unit tests